src/pool.rs

```diff
@@ -188,7 +191,25 @@ impl<Conn: Connection> PoolInner<Conn> {
                 // Handle requests from clients
                 request = self.rx.recv() => {
                     match request {
-                        Some(Request::Claim { tx }) => self.claim_or_enqueue(tx).await,
+                        Some(Request::Claim { tx }) => {
+                            self.claim_or_enqueue(tx).await
+                        }
+                        // The caller has explicitly asked us to terminate, and
+                        // we should respond to them once we've stopped doing
+                        // work.
+                        Some(Request::Terminate { tx }) => {
+                            // Terminate all background tasks, including:
+                            // - The resolver (may or may not have background
+                            // tasks, this is dependent on the implementation)
+                            // - Each of the slot sets
+                            self.terminate().await;
+                            let _ignored_result = tx.send(Ok(()));
+                            return;
+                        },
+                        // The caller has abandoned their connecion to the pool.
+                        //
+                        // We stop handling new requests, but have no one to
+                        // notify.
```

Comment:
Should we also terminate the background tasks in this case (without notifying anyone once we have finished doing so)? Otherwise, it seems like they could potentially be leaked.